staging/src/k8s.io/client-go/discovery/cached/memory/memcache.go

```diff
@@ -74,6 +75,11 @@ func isTransientError(err error) bool {
 		return true
 	}
 
+	if meta.IsNoMatchError(err) {
+		// Kind not found, API / CRD for it may be registered later.
+		return true
+	}
+
```

Comment:
I would at least expect some reasonable backoff, i.e. to not pass through yet another call to the API many times a second.

Compare, today `kubectl get doesnotexist` reloads discovery info every single time and is very very slow. We want to avoid such a behaviour in controllers.

Reply:
I think the retry policy is up to client-go user. In my app, I can see the error is returned from `myDeferredDiscoveryRESTMapper.RESTMapping()`, i.e. `DeferredDiscoveryRESTMapper` does not retry by itself. Now it's responsibility of the application to retry with exp. backoff, as with any other API server call.

Reply:
a couple of comments that may be relevant:
- do we need to consider the case if the groupVersion has been removed?
- not sure how this affects you - looks like  `NotFound` error is absorbed by the delegated discovery client https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/discovery/discovery_client.go#L203-L206

Reply:
* If a `groupVersion` was removed, it can be added later.  client-go still should not cache the error and try again if it's asked to do so.
* I am not sure what `NotFound` error is consumed in the delegated discovery client, but the memory cache gets `NoKindMatchError`, which is not a HTTP error.

Reply:
Which influence will this have to other components? Where is the memcached variant used? What about gc or namespace controllers?